apis/kubedb/v1alpha2/solr_helpers.go

```diff
@@ -262,43 +259,40 @@ func (s *Solr) SetDefaults(slVersion *catalog.SolrVersion) {
 		}
 
 		if s.Spec.PodTemplate.Spec.SecurityContext == nil {
-			s.Spec.PodTemplate.Spec.SecurityContext = &v1.PodSecurityContext{FSGroup: slVersion.Spec.SecurityContext.RunAsUser}
+			s.Spec.PodTemplate.Spec.SecurityContext = &v1.PodSecurityContext{}
 		}
 
 		s.Spec.PodTemplate.Spec.SecurityContext.FSGroup = slVersion.Spec.SecurityContext.RunAsUser
 		s.setDefaultContainerSecurityContext(slVersion, &s.Spec.PodTemplate)
-		s.setDefaultInitContainerSecurityContext(slVersion, &s.Spec.PodTemplate)
 		s.setDefaultContainerResourceLimits(&s.Spec.PodTemplate)
 	}
 }
 
-func (s *Solr) setDefaultInitContainerSecurityContext(slVersion *catalog.SolrVersion, podTemplate *ofst.PodTemplateSpec) {
+func (s *Solr) setDefaultContainerSecurityContext(slVersion *catalog.SolrVersion, podTemplate *ofst.PodTemplateSpec) {
 	initContainer := coreutil.GetContainerByName(podTemplate.Spec.InitContainers, SolrInitContainerName)
 	if initContainer == nil {
 		initContainer = &v1.Container{
 			Name: SolrInitContainerName,
 		}
+		podTemplate.Spec.InitContainers = coreutil.UpsertContainer(podTemplate.Spec.InitContainers, *initContainer)
```

Comment:
This line should be out of `If`condition. Otherwise it won't append in the first iteration when condition does not hold.

Reply:
ok

Reply:
If condition does not hold that mean there is init container info in pod template. Then do we need to append?

Reply:
It is not append rather it's upsert